app/uk/gov/hmrc/economiccrimelevyregistration/services/RegistrationValidationService.scala

```diff
@@ -326,4 +334,36 @@ class RegistrationValidationService @Inject() (clock: Clock, schemaValidator: Sc
 
   private def missingErrorMessage(missingDataDescription: String): String = s"$missingDataDescription is missing"
 
+  private def validateOtherEntity(
+    registration: Registration
+  ): ValidationResult[(String, String) => LegalEntityDetails] =
+    registration.optOtherEntityJourneyData match {
+      case Some(journey) =>
+        journey.businessName match {
+          case Some(value) if !value.isBlank =>
+            journey.entityType match {
+              case Some(Charity) => validateCharity(registration, journey)
+              case _             => ???
+            }
+          case _                             => DataValidationError(DataMissing, missingErrorMessage("business name")).invalidNel
+        }
```

Comment:
We don't need to validate the contents of strings in the backend like this, e.g. empty/blank/non empty/non blank as we validate at the point of data entry, it is overkill to do this for 'other'. In the case of the normal entity types that aren't 'other', this data is schema validated.

As you have rightly noticed - we do not need to schema validate for 'Other' as it is not a JSON payload.

The purpose of this validation is to ensure we have the minimum data set in order to make the submission. You should be able to re-use methods like `validateOptExists` and. `validateConditionalOptExists` in most cases.